File: solutions/lua/resistor-color/2/resistor-color.lua
local codes = {
	"black",
	"brown",
	"red",
	"orange",
	"yellow",
	"green",
	"blue",
	"violet",
	"grey",
	"white",
}

local solution = {}
solution.colors = function()
	return codes
end

solution.color_code = function(color)
	for i, v in ipairs(codes) do
		if v == color then
			return i - 1
		end
	end
	error("invalid color: " .. color)
end

return solution

File: solutions/elixir/resistor-color/2/lib/resistor_color.ex
defmodule ResistorColor do
  @color_codes %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  @doc """
  Return the value of a color band
  """
  @spec code(atom) :: integer()
  def code(color), do: Map.fetch!(@color_codes, color)
end
